apps/rush-lib/src/cli/actions/GenerateAction.ts

```diff
@@ -104,35 +109,37 @@ export default class GenerateAction extends BaseRushAction {
       fsx.unlinkSync(tempShrinkwrapFilename);
     }
 
-    if (isLazy) {
-      console.log(colors.green(
-        `${os.EOL}Rush is running in "--lazy" mode. ` +
-        `You will need to run a normal "rush generate" before committing.`));
+    if (isLazy || this.rushConfiguration.packageManager === 'pnpm') {
```

Comment:
Can we have one class for pnpm and one class for npm for generate? It will be cleaner than having a check for pnpm all over this class?

Reply:
That seems more difficult. I just split them into more independent code blocks that have just a small replication of code.

---
In reply to: [153378445](https://github.com/Microsoft/web-build-tools/pull/449#discussion_r153378445) [](ancestors = 153378445)